### ISSUE TITLE
Bash files syntax highlight

### ISSUE THREAD

**Issue by surveyor3** [geany/geany#1821]
In geany .sh file have a syntax highlighting that's not quite good, compared  i.e. to other editors like sublime. In particular when you have $(...) things inside the parenthesis are all colored in orange.
Geany version: 1.31
Color combination: default
GTK version: 2.24.31
O.S.: Ubuntu 17.10
filetype: .sh

**Comment by elextr** [comment]
If you don't like the colours you can choose another colour scheme from those [contributed by others](https://github.com/geany/geany-themes)  or make your own.

**Comment by b4n** [comment]
@elextr I think @surveyor3 is complaining that the syntax inside of `$()` is not highlighted specifically, but only as a single `$()` block.  In shell, `$()` is a command substitution, so the content inside it could be highlighted as normal syntax.

This is a Scintilla lexer limitation, see https://sourceforge.net/p/scintilla/feature-requests/1033/

**Comment by elextr** [comment]
>  In shell, $() is a command substitution, so the content inside it could be highlighted as normal syntax.

Oh, I thought it was just the orange that was the problem, ok.

**Comment by surveyor3** [comment]
Yes, it's not about the orange but is about the thing that inside is all highlighted in orange and it's not correct. 
I did a screenshot to show you the issue.
![schermata del 2018-04-09 23-45-20](https://user-images.githubusercontent.com/21111910/38524778-164c3b50-3c50-11e8-9329-3ba9cada242b.png)

Thanks for the attention and for all the good work in Geany.

**Comment by LarsGit223** [comment]
I posted a PR in the scintilla project to fix this, see https://sourceforge.net/p/scintilla/code/merge-requests/22.
